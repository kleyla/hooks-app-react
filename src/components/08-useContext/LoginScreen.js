import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  return (
    <div>
      <h1>Login</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() =>
          setUser({
            id: 123,
            name: "karen",
          })
        }
      >
        Login
      </button>

      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};
