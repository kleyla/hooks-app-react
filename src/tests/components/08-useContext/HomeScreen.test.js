import { mount } from "enzyme";
import { HomeScreen } from "../../../components/08-useContext/HomeScreen";
import { UserContext } from "../../../components/08-useContext/UserContext";

describe("Pruebas en HomeScreen", () => {
  const user = {
    name: "Karen",
    email: "Ley@live.com",
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test("Debe de mostrarse correctamete", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
