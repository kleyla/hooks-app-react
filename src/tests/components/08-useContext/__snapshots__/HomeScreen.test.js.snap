// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen Debe de mostrarse correctamete 1`] = `
<div>
  <h1>
    Home
  </h1>
  <hr />
  <pre>
    {
   "name": "Karen",
   "email": "Ley@live.com"
}
  </pre>
</div>
`;
